### PR TITLE
Generate: generation config validation fixes in docs

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

Fixes a few failing doctests as a result of #25381

### CODE REVIEW

File: docs/source/en/model_doc/donut.md [hunk 1]
```diff
@@ -80,11 +80,9 @@ into a single instance to both extract the input features and decode the predict
 ...     pixel_values.to(device),
 ...     decoder_input_ids=decoder_input_ids.to(device),
 ...     max_length=model.decoder.config.max_position_embeddings,
-...     early_stopping=True,
```
Comment (gante) [inline docs/source/en/model_doc/donut.md:83]: early_stopping is only used with beam methods (num_beams>1)

File: docs/source/en/model_doc/donut.md [hunk 2]
```diff
@@ -80,11 +80,9 @@ into a single instance to both extract the input features and decode the predict
...
 ...     pad_token_id=processor.tokenizer.pad_token_id,
 ...     eos_token_id=processor.tokenizer.eos_token_id,
 ...     use_cache=True,
-...     num_beams=1,
```
Comment (gante) [inline docs/source/en/model_doc/donut.md:87]: this is the default, can be removed

File: src/transformers/generation/configuration_utils.py [hunk 3]
```diff
@@ -597,7 +597,7 @@ def from_pretrained(
         >>> # If you'd like to try a minor variation to an existing configuration, you can also pass generation
         >>> # arguments to `.from_pretrained()`. Be mindful that typos and unused arguments will be ignored
         >>> generation_config, unused_kwargs = GenerationConfig.from_pretrained(
-        ...     "gpt2", top_k=1, foo=False, return_unused_kwargs=True
+        ...     "gpt2", top_k=1, foo=False, do_sample=True, return_unused_kwargs=True
```
Comment (gante) [inline src/transformers/generation/configuration_utils.py:600]: top_k requires do_sample=True

Comment (ydshieh) [reply src/transformers/generation/configuration_utils.py:600]: would this change the output?

**Review by ydshieh — APPROVED** [review]
Thanks! Just a tiny question about if the output would change (if so, we need to update?)

File: src/transformers/generation/configuration_utils.py (hunk 3 above)
Comment (gante) [reply src/transformers/generation/configuration_utils.py:600]: No, because this doctest only checks other fields :)

**Review by sgugger — APPROVED** [review]
Thanks for fixing!

**Pull request merged**